### PR TITLE
Add package 'docker:image-classification-tf-py-ubuntu-18.04' with CUDA

### PR DESCRIPTION
**Pull request by bellycat77**
- add CUDA support to image

### CODE REVIEW

File: docker/image-classification-tf-py-ubuntu-18.04/Dockerfile.cuda [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+# Download and preprocess the first 500 images of the ImageNet 2012 validation dataset.
+RUN ck install package --tags=dataset,imagenet,aux
+RUN ck install package --tags=dataset,imagenet,val,original,min --no_tags=resized
+RUN ck install package --tags=dataset,imagenet,val,preprocessed
+
+# Download MobileNet TF model.
+RUN ck install package --tags=image-classification,model,tf,mlperf,mobilenet,non-quantized
+
+# Run Image Classification TFLite program once.
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/Dockerfile.cuda:87]: TFLite -> TF

File: docker/image-classification-tf-py-ubuntu-18.04/Dockerfile.cuda [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#RUN ck install package --tags=lib,python-package,scipy
+RUN ${CK_PYTHON} -m pip install scipy==1.2.1 --user && ck detect soft:lib.python.scipy
+RUN ck install package --tags=lib,python-package,pillow
+
+# Install C/C++ dependencies.
+RUN ck detect soft:compiler.gcc --full_path=`which ${CK_CC}`
+
+# Install CK packages for tensorflow
+RUN ck detect soft:compiler.cuda --full_path=/usr/local/cuda-10.0/bin/nvcc \
+ && ck detect soft:lib.cudnn \
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/Dockerfile.cuda:75]: How about the path for cuDNN?

File: docker/image-classification-tf-py-ubuntu-18.04/Dockerfile.cuda [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+# Use generic Linux settings with dummy frequency setting scripts.
+RUN ck detect platform.os --platform_init_uoa=generic-linux-dummy
+
+# Install Python dependencies.
+RUN ${CK_PYTHON} -m pip install --ignore-installed pip setuptools --user
+RUN ck detect soft:compiler.python --full_path=`which ${CK_PYTHON}`
+
+RUN ck install package --tags=lib,python-package,numpy
+#RUN ck install package --tags=lib,python-package,scipy
+RUN ${CK_PYTHON} -m pip install scipy==1.2.1 --user && ck detect soft:lib.python.scipy
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/Dockerfile.cuda:67]: Use CK installer with fixed versions?

File: docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+**NB:** `#` means execution under root or with `sudo`.
+
+### Change directory
+```
+$ cd `ck find docker:image-classification-tf-py-ubuntu-18.04`
+```
+
+### Prepare
+Due to NVIDIA restrictions, You should register (free) at https://developer.nvidia.com/
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md:11]: You -> you

File: docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+ After that You should create `cuda.tar` with following structure:
+```
+cuda.tar:
+ /cuda
+   /lib
+     Ubuntu18_04-x64/libcudnn7_7.5.1.10-1+cuda10.0_amd64.deb
+   /lib-dev
+     Ubuntu18_04-x64/libcudnn7-dev_7.5.1.10-1+cuda10.0_amd64.deb
+```
+and put `cuda.tar` in current directory
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md:27]: in the directory returned by `ck find docker:image-classification-tf-py-ubuntu-18.04`?

File: docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+### Prepare
+Due to NVIDIA restrictions, You should register (free) at https://developer.nvidia.com/
+for [downloading](https://developer.nvidia.com/rdp/cudnn-download) required `cuDNN` packages.
+
+You should download cuDNN v7.5.1 for CUDA 10.0:
+ - [cuDNN Runtime Library for Ubuntu18.04 (Deb)](https://developer.nvidia.com/compute/machine-learning/cudnn/secure/v7.5.1/prod/10.0_20190418/Ubuntu18_04-x64/libcudnn7_7.5.1.10-1%2Bcuda10.0_amd64.deb)
+ - [cuDNN Developer Library for Ubuntu18.04 (Deb)](https://developer.nvidia.com/compute/machine-learning/cudnn/secure/v7.5.1/prod/10.0_20190418/Ubuntu18_04-x64/libcudnn7-dev_7.5.1.10-1%2Bcuda10.0_amd64.deb)
+
+ After that You should create `cuda.tar` with following structure:
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md:18]: `cuda.tar` -> `cudnn.tar`

**Comment by psyhtest** [comment]
@bellycat77 By the way, please see if you can draw some inspiration from @guschmue's [Dockerfile.gpu](https://github.com/mlperf/inference/blob/master/v0.5/classification_and_detection/Dockerfile.gpu). (I don't see him installing cuDNN though.)

**Comment by guschmue** [comment]
No cudnn install because it comes from the nvidia image:
```
FROM nvidia/cuda:10.0-cudnn7-devel-ubuntu16.04
```

**Comment by psyhtest** [comment]
Thanks @guschmue!

@bellycat77 Could you please try adding another CK entry e.g. `docker:image-classification-tf-py.nvidia-cuda10.1-cudnn7-devel-ubuntu18.04` using:
```
FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu18.04
```
from the [NVIDIA Docker hub](https://hub.docker.com/r/nvidia/cuda/).

File: docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+   /lib
+     Ubuntu18_04-x64/libcudnn7_7.5.1.10-1+cuda10.0_amd64.deb
+   /lib-dev
+     Ubuntu18_04-x64/libcudnn7-dev_7.5.1.10-1+cuda10.0_amd64.deb
+```
+and put `cuda.tar` in current directory
+
+### Build image
+```
+# docker build . -f Dockerfile.cuda -t image-classification-tf-py-ubuntu-18.04
```
Comment (psyhtest) [inline docker/image-classification-tf-py-ubuntu-18.04/README.CUDA.md:31]: Please check the latest master. We now prefer to use:
```
docker build -f Dockerfile.cuda -t image-classification-tf-py.ubuntu-18.04 .
```
(the current path `.` at the end; dot between the program name and the base image name).

**Comment by psyhtest** [comment]
We now base our Docker images that require CUDA support on NVIDIA image e.g. see [`docker:object-detection-tf-py.tensorrt.ubuntu-18.04`](https://github.com/ctuning/ck-object-detection/tree/master/docker/object-detection-tf-py.tensorrt.ubuntu-18.04)

**Pull request closed without merging**